vizro-core/src/vizro/models/_components/ag_grid.py

```diff
@@ -113,7 +114,7 @@ def build(self):
         return dcc.Loading(
             [
                 html.H3(self.title, className="table-title") if self.title else None,
-                html.Div(self.figure._function(**dash_ag_grid_conf), id=self.id, className="table-container"),
+                html.Div(grid, id=self.id, className="table-container"),
```

Comment:
So the reason why we did it for the AgGrid was the pagination - and then the worry that there may be other arguments we do not know about which might react differently.

We never did it for the Dash DataTable because there were no problems observed. I would say, let's write the tests for the dash table, but not necessarily change the implementation until we need to?

Reply:
Fine by me 😄

Reply:
I'd like to understand this a bit better because both the `Table.build` and `AgGrid.build` approach seem a bit weird to me, but maybe there's a good reason for it.

Either way happy to approve this for now and discuss in a different PR.

Reply:
ok dokes, let's discuss this later.

This follows the approach we discussed and you suggested for removing the `id` from the pre-build.

The build method then aims to create an object with that id so that callbacks etc register properly. In the case of the AgGrid it is important that the object (which then gets overwritten by the on page load) has the same settings - having different pagination settings caused trouble before